controllers/component_controller_unit_test.go

```diff
@@ -348,3 +350,113 @@ func TestGenerateGitops(t *testing.T) {
 	}
 
 }
+
+func TestGetGitProvider(t *testing.T) {
+	type args struct {
+		ctx    context.Context
+		gitURL string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantErr    bool
+		wantString string
+	}{
+		{
+			name: "github",
+			args: args{
+				ctx:    context.Background(),
+				gitURL: "git@github.com:redhat-appstudio/application-service.git",
+			},
+			wantErr:    true, //unsupported
+			wantString: "",
+		},
+		{
+			name: "github https",
+			args: args{
+				ctx:    context.Background(),
+				gitURL: "https://github.com/redhat-appstudio/application-service.git",
+			},
+			wantErr:    false,
+			wantString: "https://github.com",
+		},
+		{
+			name: "bitbucket https",
+			args: args{
+				ctx:    context.Background(),
+				gitURL: "https://sbose78@bitbucket.org/sbose78/appstudio.git",
+			},
+			wantErr:    false,
+			wantString: "https://bitbucket.org",
+		},
+		{
+			name: "no scheme",
```

Comment:
Nit: Can you add a quick test for an invalid git url / something that causes url.Parse to fail.

It'll get codecov to complain about two less lines 🙂